### PR TITLE
Models-TypeSpec Remove and Test

### PR DESCRIPTION
**Pull request by mcgallan**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/AutoRest.TestServer.Tests/ModelsTypeSpecTests.cs [hunk 1]
```diff
@@ -424,17 +424,6 @@ public void InputCollectionPropertiesCanBeMutatedAfterConstruction()
             Assert.AreEqual(1, inputModel.RequiredIntList.Count);
         }
 
-        [Test]
-        public void ReadOnlyPropertiesAreReadOnly()
-        {
-            var property = TypeAsserts.HasProperty(typeof(RoundTripReadOnlyModel), "RequiredReadonlyString", BindingFlags.Public | BindingFlags.Instance);
-            var listProperty = TypeAsserts.HasProperty(typeof(RoundTripReadOnlyModel), "RequiredReadOnlyModelList", BindingFlags.Public | BindingFlags.Instance);
-
-            Assert.Null(property.SetMethod);
```
Comment (pshao25) [inline test/AutoRest.TestServer.Tests/ModelsTypeSpecTests.cs:433]: Where are these tests now?

File: test/AutoRest.TestServer.Tests/LowLevel/ModelsTypeSpecTests.cs [hunk 2]
```diff
@@ -1,63 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
```
Comment (pshao25) [inline test/AutoRest.TestServer.Tests/LowLevel/ModelsTypeSpecTests.cs:4]: You cannot remove these. Add them to cadlranchprojects.tests/models-typespec.ts

File: test/TestProjects/Models-TypeSpec/Models-TypeSpec.tsp [hunk 3]
```diff
@@ -390,150 +357,6 @@ model OutputModel {
...
-
-  @doc("Optional collection of which the element is a nullable int")
-  optionalCollectionWithNullableIntElement?: (int32 | null) [];
-}
-
-@doc("Output model with readonly properties.")
-model RoundTripReadOnlyModel {
-  @doc("Required string, illustrating a readonly reference type property.")
-  @visibility("read")
-  requiredReadonlyString: string;
```
Comment (pshao25) [inline test/TestProjects/Models-TypeSpec/Models-TypeSpec.tsp:446]: All these properties have `@visibility`. Does cadl-ranch have corresponding ones?

**Review by pshao25 — CHANGES_REQUESTED** [review]
You cannot delete any test that is not e2e, i.e., local tests without cadl-ranch involved should keep.

File: test/TestProjects/Models-TypeSpec/Models-TypeSpec.tsp (hunk 3 above)
Comment (mcgallan) [reply test/TestProjects/Models-TypeSpec/Models-TypeSpec.tsp]: fixed

File: test/AutoRest.TestServer.Tests/LowLevel/ModelsTypeSpecTests.cs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (pshao25) [inline test/AutoRest.TestServer.Tests/LowLevel/ModelsTypeSpecTests.cs:1]: Others looks good! Let's revert this change and merge the latest code then we are good to go.

Comment (mcgallan) [reply test/AutoRest.TestServer.Tests/LowLevel/ModelsTypeSpecTests.cs:1]: fixed

**Review by pshao25 — APPROVED** [review]
